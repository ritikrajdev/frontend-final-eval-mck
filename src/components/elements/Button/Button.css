.custom-button {
  min-width: 150px;
  padding: 10px;
  border: none;
  border-radius: 8px;
  background-color: var(--dark-purple);
  color: white;
  cursor: pointer;
}
