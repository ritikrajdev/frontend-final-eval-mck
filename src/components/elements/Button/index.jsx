import PropTypes from 'prop-types';
import React from 'react';

import './Button.css';

export default function Button({ children, type }) {
  return (
    <button className='custom-button' type={type}>
      {children}
    </button>
  );
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.string,
};
