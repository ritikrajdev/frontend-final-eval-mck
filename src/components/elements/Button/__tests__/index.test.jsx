import { render } from '@testing-library/react';
import React from 'react';
import Button from '..';

describe('Button', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button>Click me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render a button', () => {
    const { container } = render(<Button>Click me</Button>);
    expect(container.querySelector('button')).toBeInTheDocument();
  });
  it('should render a button with the correct text', () => {
    const { container } = render(<Button>Click me</Button>);
    expect(container.querySelector('button')).toHaveTextContent('Click me');
  });
  it('should render a button with the correct type', () => {
    const { container } = render(<Button type='submit'>Click me</Button>);
    expect(container.querySelector('button')).toHaveAttribute('type', 'submit');
  });
});
