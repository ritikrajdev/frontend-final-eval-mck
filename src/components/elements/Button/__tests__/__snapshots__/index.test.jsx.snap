// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<DocumentFragment>
  <button
    class="custom-button"
  >
    Click me
  </button>
</DocumentFragment>
`;
